DataStructure/src/main/list/ResizableArray.java

```diff
@@ -0,0 +1,167 @@
+package main.list;
+
+/**
+ * Represents a resizable array of integer elements.
+ */
+
+public class ResizableArray {
+    private int size;
+    private int firstBlankIndex;
+    private int list[];
+
+    /**
+     * Constructs a resizable array
+     * @param size
+     * size of the list
+     */
+    public ResizableArray(int size){
+        this.size = size;
+        list = new int[this.size] ;
+        firstBlankIndex = 0;
+    }
+
+    /**
+     * Constructs a resizable array with default size 50
+     */
+    public ResizableArray(){
+        size = 50;
+        list = new int[50] ;
+        firstBlankIndex = 0;
+    }
+
+    /**
+     * Adds an element to the list
+     * @param num
+     */
+    public void add(int num){
+        if (size == 0){
+            list[size] = num;
+            return;
+        }
+        else if (firstBlankIndex == size){
+            size = size*2 ;
+            int list2 [] = new int[size];
+            for (int i = 0; i < firstBlankIndex; i++) {
+               list2[i] = list[i];
+            }
+            list = list2;
+        }
+        list[firstBlankIndex] = num;
+        firstBlankIndex++;
+    }
+
+    /**
+     * Get element at the given index.
+     *@param index
+     * @return
+     * Returns the element
+     */
+    public int get(int index){
+        if (index < 0 || index >= firstBlankIndex ){ return -1;}
+        return list[index];
+    }
+
+    /**
+     * Sets element at specified index
+     *
+     * @param index
+     * @param num
+     */
+    public void set(int index, int num) {
+        if (index < 0 || index >= firstBlankIndex ){ return;}
+        list[index] = num;
+    }
+
+    /**
+     * Removes last element and returns it
+     */
+    public int remove(){
+        if (firstBlankIndex == 0 || size == 0 ){ return -1;}
```

Comment:
There is no need to check for size == 0. 
You can call isEmpty method to make the decision. 
The method mentions that returns last element but you are returning -1 when resizable array is empty. 
In which scenarios, the array could become empty ?
Why it would be good idea to throw an exception if the array is empty ?